src/lib/y2partitioner/actions/controllers/filesystem.rb

```diff
@@ -137,65 +152,49 @@ def apply_role
           delete_filesystem
           @encrypt = false
 
-          fs_type = mount_point = mount_by = nil
-
-          case role
-          when :swap
-            part_id = Y2Storage::PartitionId::SWAP
-            fs_type = Y2Storage::Filesystems::Type::SWAP
-            mount_point = "swap"
-            mount_by = Y2Storage::Filesystems::MountByType::DEVICE
-          when :efi_boot
-            part_id = Y2Storage::PartitionId::ESP
-            fs_type = Y2Storage::Filesystems::Type::VFAT
-            mount_point = "/boot/efi"
-          when :raw
-            part_id = Y2Storage::PartitionId::LVM
-          else
-            part_id = DEFAULT_PARTITION_ID
-            fs_type = (role == :system) ? DEFAULT_FS : DEFAULT_HOME_FS
-          end
+          self.partition_id = role_values[:partition_id]
 
-          self.partition_id = part_id
+          fs_type = role_values[:filesystem_type]
+          mount_path = role_values[:mount_path]
+          mount_by = role_values[:mount_by]
 
-          return unless fs_type
+          return if fs_type.nil?
 
           create_filesystem(fs_type)
-          assign_fs_attrs(mount_by: mount_by)
-          self.mount_point = mount_point
+          create_mount_point(mount_path, mount_by: mount_by) unless mount_path.nil?
         end
 
         # Creates a new filesystem on top of the block device, removing the
         # previous one if any, as a result of the user choosing the option in the
         # UI to format the device.
         #
-        # Some information from the previous filesystem (like the mountpoint or
+        # Some information from the previous filesystem (like the mount point or
         # the label) is kept in the new filesystem if it makes sense.
         #
-        # @param type of the new filesystem
+        # @param type [Symbol] e.g., :ext4, :btrfs, :xfs
         def new_filesystem(type)
           # Make sure type has the correct... well, type :-)
           type = Y2Storage::Filesystems::Type.new(type)
 
           # It's kind of expected that these attributes are preserved when
           # changing the filesystem type, with the exceptions below
-          mount_point = current_value_for(:mount_point)
+          mount_path = current_value_for(:mount_point)
           mount_by = current_value_for(:mount_by)
           label = current_value_for(:label)
 
           if type.is?(:swap)
-            mount_point = "swap"
-          elsif mount_point == "swap"
-            mount_point = ""
+            mount_path = "swap"
+          elsif mount_path == "swap"
```

Comment:
This part I don't understand. I can see that it was very much like this before, but still, does this really work?
If we have a type :swap, we set a mount_path to "swap". But if it was not :swap, but already had a mount_path "swap", the mount_path is reset to nil. Isn't that contradictory? Shouldn't this behave the same way in both cases - when the type is :swap or when the mount_path is "swap"?

Reply:
Well, the idea is always set mount_path to "swap" when selected fs is swap. In case that selected fs is other different to swap, we leave the current mount_path (if there is one). In this last case, we should ensure that the reused mount path is not "swap", because that does not make sense.

Reply:
OK.